soc/arm/nordic_nrf/nrf53/Kconfig.soc

```diff
@@ -224,3 +224,7 @@ config NRF_ENABLE_CACHE
 
 	  Instruction cache only (I-Cache) is available in nRF5340
 	  CPUNET (Network MCU).
+
+config BUILD_WITH_TFM
+	# TF-M nRF53 platform enables the cache unconditionally.
+	select NRF_ENABLE_CACHE if SOC_NRF5340_CPUAPP
```

Comment:
NRF_ENABLE_CACHE will be shown in the menu as something that can be disabled.

Let's make the prompt conditional on ! BUILD_WITH_TFM.

Then it will default to y and it will not be possible to disable it, nor will the user be presented with a means to disable it.

config NRF_ENABLE_CACHE
bool
prompt "enable cache" if ! BUILD_WITH_TFM

Reply:
No, with this way it will show up as selected and not be able to de-select it in menuconfig.
Also you get this:
```
west build samples/tfm_integration/tfm_regression_test -- -DCONFIG_NRF_ENABLE_CACHE=n
warning: NRF_ENABLE_CACHE (defined at soc/arm/nordic_nrf/nrf53/Kconfig.soc:216) was assigned the
value 'n' but got the value 'y'.
```

With the conditional prompt it disappears from the menu as well, but otherwise I think they behave the same.

Reply:
My mistake, I thought you could disable in menuconfig something that was selected.

It's still marginally cleaner to not present the user with a checkbox he can't check.

But this is not that important, no action is OK.